github_poster/loader/neodb_loader.py

```diff
@@ -0,0 +1,84 @@
+import requests
+
+from github_poster.loader.base_loader import BaseLoader, LoadError
+from github_poster.loader.config import NEODB_API
+
+
+class NeoDBLoader(BaseLoader):
+    unit = "Marks"
+    # track_color = "#78C800"
+
+    def __init__(self, from_year, to_year, _type, **kwargs):
+        super().__init__(from_year, to_year, _type, **kwargs)
+        self.neodb_token = kwargs.get("neodb_token", "")
+        self.mark_type = kwargs.get("mark_type", "complete")
+
+    @classmethod
+    def add_loader_arguments(cls, parser, optional):
+        parser.add_argument(
+            "--neodb_token",
+            dest="neodb_token",
+            type=str,
+            help="Get token from https://neodb.social/developer",
+            required=optional,
+        )
+        parser.add_argument(
+            "--mark_type",
+            dest="mark_type",
+            type=str,
+            help="The type of marks you want to show, like 'complete', 'wishlist', 'progress'",
+            required=optional,
+            default="complete",
+        )
+
+    def get_api_data(self):
+        page = 1
+        count = 0
+
+        try:
+            headers = {
+                "Authorization": f"Bearer {self.neodb_token}",
+                "Content-Type": "application/json",
+            }
+
+            while True:
+                r = requests.get(
+                    url=NEODB_API.format(page=page, type=self.mark_type),
+                    headers=headers,
+                )
+
+                all_count = r.json().get("count")
+                all_page = r.json().get("pages")
+
+                data_list = r.json().get("data")
```

Comment:
give them a default value since you use dict.get maybe better?

Reply:
and I think all_page and all_count only we can get it once?  the other times all the same?

Reply:
fixed in the latest commit